List match percentage of all saved fingers

deactivated-acc opened the pull request:
Closes #78

deactivated-acc committed:
List match percentage of all saved fingers

jrauhamaa commented on FingerBlox/app/src/main/java/org/fingerblox/fingerblox/ImageDisplayActivity.java:
```diff
@@ -156,26 +159,37 @@ protected void saveFeatures(String fileName){
         }
 
         AlertDialog.Builder builder = new AlertDialog.Builder(this);
         builder.setTitle("Match found");
```
If fileNameList is empty, title shouldn't be "Match found"

deactivated-acc replied:
Before I added the listview it would also always show match found. Can change to match not found if no files present.

jrauhamaa replied:
That would be good. Something like "cannot match without saved images" would be even better.

jorenham commented on FingerBlox/app/src/main/java/org/fingerblox/fingerblox/ImageDisplayActivity.java:
```diff
@@ -156,26 +159,37 @@ protected void saveFeatures(String fileName){
     public void findMatch() {
         SharedPreferences preferences = getSharedPreferences("PREFS", 0);
         String[] fileNameList = preferences.getString("fileNameList", "").split(" ");
+        ArrayList<Pair<String, Integer>> matches = new ArrayList<>();
```
Why not a `TreeMap<Integer, String>`? That way it would be automatically sorted and the code would be cleaner

deactivated-acc replied:
Because the adapter for the listview uses an (array)list. The adapter also needs to be able to get the items in a particular way to be able to display them.

jorenham commented on FingerBlox/app/src/main/java/org/fingerblox/fingerblox/ImageDisplayActivity.java:
```diff
@@ -156,26 +159,37 @@ protected void saveFeatures(String fileName){
         builder.setPositiveButton("OK", null);
+
+        LayoutInflater inflater = getLayoutInflater();
+        View dialogView = inflater.inflate(R.layout.match_dialog, null);
```
What's going on here?

deactivated-acc replied:
This is how I learned how to set the layout for the dialog, with using the builder and such. If there is a better way, please tell me.

jorenham commented on FingerBlox/app/src/main/java/org/fingerblox/fingerblox/ImageDisplayActivity.java:
```diff
@@ -156,26 +159,37 @@ protected void saveFeatures(String fileName){
         AlertDialog dialog = builder.show();
 
+        ListView matchListView = (ListView) dialog.findViewById(R.id.dialog_listview);
+        MatchAdapter matchAdapter = new MatchAdapter(getApplicationContext(), R.layout.match_row_item, matches);
```
Passing `this` instead of `getApplicationContext()` would be better

deactivated-acc replied:
You're right.

jorenham commented on FingerBlox/app/src/main/java/org/fingerblox/fingerblox/MatchAdapter.java:
```diff
@@ -0,0 +1,50 @@
+import android.widget.TextView;
+import java.util.List;
+
+public class MatchAdapter extends ArrayAdapter<Pair<String, Integer>> {
```
package-protected would be better than `public`

deactivated-acc replied:
Also right.

jorenham reviewed (changes requested):
Couple of remarks. Also I'm not sure what you need all the inflation for?

deactivated-acc committed:
Some fixes to list all matches

-Made matchadapter private
-Replaced getapplicationcontext with this
-Show different dialog when there are no files to match with
